astropy/coordinates/builtin_frames/lsr.py

```diff
@@ -158,3 +159,93 @@ def galacticlsr_to_galactic(lsr_coord, galactic_frame):
     v_offset = v_bary_gal.data.represent_as(r.CartesianDifferential)
     offset = r.CartesianRepresentation([0, 0, 0]*u.au, differentials=-v_offset)
     return None, offset
+
+
+# ------------------------------------------------------------------------------
+
+# The LSRK velocity frame, defined as having a velocity of 20 km/s towards
+# RA=270 Dec=30 (B1900) relative to the solar system Barycenter. This is defined
+# in:
+#
+#   Gordon 1975, Methods of Experimental Physics: Volume 12:
+#   Astrophysics, Part C: Radio Observations - Section 6.1.5.
+
+
+class LSRK(BaseRADecFrame):
+    r"""
+    A coordinate or frame in the Kinematic Local Standard of Rest (LSR).
+
+    This frame is defined as having a velocity of 20 km/s towards RA=270 Dec=30
+    (B1900) relative to the solar system Barycenter. This is defined in:
+
+        Gordon 1975, Methods of Experimental Physics: Volume 12:
+        Astrophysics, Part C: Radio Observations - Section 6.1.5.
+
+    This coordinate frame is axis-aligned and co-spatial with `ICRS`, but has
+    a velocity offset relative to the solar system barycenter to remove the
+    peculiar motion of the sun relative to the LSRK.
+    """
+
+
+GORDON1975_V_BARY = 20*u.km/u.s
+GORDON1975_DIRECTION = FK4(ra=270*u.deg, dec=30*u.deg, equinox='B1900')
+
+v_offset_lsrk = ((GORDON1975_V_BARY * GORDON1975_DIRECTION.transform_to(ICRS).data)
```

Comment:
In fact it would actually be better to simply compute all the offsets in ICRS and hard-code those (not even the Galactic values). Maybe what I'll do is put the final values hard-coded, but leave the code to generate them in comments for reproducibility.

Reply:
Done